Lib/test/test_exception_variations.py

```diff
@@ -172,6 +172,127 @@ def test_nested_else(self):
         self.assertTrue(hit_finally)
         self.assertTrue(hit_except)
 
+    def test_nested_exception_in_except(self):
+        hit_else = False
+        hit_finally = False
+        hit_except = False
+        hit_inner_except = False
+        hit_inner_else = False
+
+        try:
+            try:
+                raise Exception('inner exception')
+            except:
+                hit_inner_except = True
+                raise Exception('outer exception')
+            else:
+                hit_inner_else = True
+        except:
+            hit_except = True
+        else:
+            hit_else = True
+        finally:
+            hit_finally = True
+
+        self.assertTrue(hit_inner_except)
+        self.assertFalse(hit_inner_else)
+        self.assertFalse(hit_else)
```

Comment:
Would these tests be slightly easier to read if the `self.assertFalse`s were inlined as `self.fail()`? It could be easier to visualise the correct path with just a short glance.

Reply:
You could do that for the assertFalse, but not for the assertTrue.

Reply:
Yes, that was the point :)

Reply:
IMO, readability is more important than following the existing style. But, I'm nitpicking. It's your pr :)